spring-integration-ws/src/main/java/org/springframework/integration/ws/AbstractWebServiceOutboundGateway.java

```diff
@@ -165,6 +165,12 @@ public void setFaultMessageResolver(FaultMessageResolver faultMessageResolver) {
 		this.webServiceTemplate.setFaultMessageResolver(faultMessageResolver);
 	}
 
+	/**
+	 * Specify the {@link WebServiceMessageSender} to use.
+	 * @param messageSender the sender.
+	 * @deprecated in favor of {@link #setMessageSenders(WebServiceMessageSender...)}
+	 */
+	@Deprecated
 	public void setMessageSender(WebServiceMessageSender messageSender) {
 		Assert.state(!this.webServiceTemplateExplicitlySet,
```

Comment:
Would you mind to change logic then in favor of calling a replacing API instead?
Then it is going to be much easier to support possible changes in the target logic afterward.

Reply:
Sorry, I missed this; but, that said, I don't understand it.

Reply:
I mean this:
```
public void setMessageSender(WebServiceMessageSender messageSender) {
   setMessageSenders(messageSender);
}
```

So, we don't need to have that `Assert` in this method any more. The logic from a deprecated method is fully delegated to replacing API.

Reply:
Doh; ok.